### ISSUE TITLE
Invalid "destination doesn't support AMP payments" error

### ISSUE THREAD

**Issue by kevkevinpal** [lightningnetwork/lnd#6431]
### Background

When trying to send an amp payment with `sendPaymentV2` if we include values in for `payment_hash`, `dest_custom_records`, and `dest_features` we get an error of `destination doesn't support AMP payments` which is misleading since the issue is the parameters being sent in the amp payment.

Below is where we found out about this and resolved the issue on our end
https://github.com/stakwork/sphinx-relay/pull/362/files#r853329296

### Your environment

* version of `lnd`
lightninglabs/lnd:v0.14.3-beta.rc1
* which operating system (`uname -a` on *Nix)
Ubuntu gitactions
* version of `btcd`, `bitcoind`, or other backend
bitcoind
* any other relevant environment details
This is the full docker-compose.yml we run for this setup
https://github.com/stakwork/sphinx-stack/blob/master/docker-compose.yml

### Steps to reproduce
you can follow this PR and you should be able to see it, it is in the integration tests that it is occurring

### Expected behaviour
It should instead tell us that we need to remove those options from the amp payment

### Actual behaviour

Instead it is giving us the `destination doesn't support AMP payments`  error

let me know if this looks like something that needs to be updated or if you have any questions :)

**Comment by Roasbeef** [comment]
Does the destination have the AMP feature bit advertised?

**Comment by kevkevinpal** [comment]
> Does the destination have the AMP feature bit advertised?

When I run `lncli getinfo` I'm seeing in the feature array 30 and 31 feature bits is that what you're referring to?

**Comment by Roasbeef** [comment]
Yes those are the feature bits. Do you see the same behavior if you use an invoice? Are you using a hop hint? If so are you setting these feature bits in the hop hint as well?
